Add method 'hasAccess' to AdminInterface. For next major release.

tim96 opened the pull request:
Add method 'hasAccess' to AdminInterface.

tim96 commented:
@OskarStark Can I ask you a question about milestone? Why milestone is 3.0? 3.0 is current branch @Soullivaneuh write it here #3731.

greg0ire commented:
I think it is not normal, I fixed that.

OskarStark commented on Admin/Admin.php:
```diff
@@ -2964,6 +2964,27 @@ public function checkAccess($action, $object = null)
     /**
      * {@inheritdoc}
      */
+    public function hasAccess($action, $object = null)
```
this method was already merged, or am i wrong?

greg0ire replied:
You're right, @tim96 should rebase on the tip of master, and it will disappear from the diff.

tim96 replied:
You are right. I will rebase this branch and fixed it.

OskarStark replied:
👍

tim96 replied:
Fixed

Soullivaneuh commented:
Please fill your PR of missing parts indicated by the "Needs" labels.

Feel free to ask if you want some precision of what is needed.

tim96 commented:
@Soullivaneuh hi. can you explain about 'Upgrade note'?

greg0ire commented:
> hi. can you explain about 'Upgrade note'?

Having a BC-break means that users are supposed to change their code when they upgrade to a new major version. The upgrade note should explain how they are supposed to change their code.

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

tim96 commented:
@Soullivaneuh @SonataCI @greg0ire Do you have roadmap or date for the next major release? And I will rebase this PR the week before this date.

greg0ire commented:
> @Soullivaneuh @SonataCI @greg0ire Do you have roadmap or date for the next major release? And I will rebase this PR the week before this date. 

Why? We will not wait until this date to merge your PR, even it contains a BC-break. This is why we have two branches.

As per the roadmap, it is not clearly defined, I think we will release if at least one of this condition is fulfilled : 

1. the previous major release is 6 month old
2. there are a great number (to be defined) of commits in master that are not in stable

But I think in the case of 4.0, the release could come earlier than in 6 months, don't recall the reason though.

tim96 commented:
@greg0ire We have two branches?

> Why?

Please, don't think that I am lazy, but I rebased this PR two times already. Tomorrow you again add something and I need to rebase it again.

Soullivaneuh commented:
@tim96 Note we just need the rebase, the upgrade note and the changelog note as requested before and the PR will be merged ASAP, even if the next major is not for now.

For the changelog, you can add it on the PR body, following this template: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/.github/PULL_REQUEST_TEMPLATE.md

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

greg0ire commented:
I know it's hard but the more you wait the more conflicts you will, since others are working. Not touching files everybody changes should help though, so following @Soullivaneuh 's advice about the CHANGELOG is the way to go. Sorry if it's cumbersome.

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

tim96 commented:
@Soullivaneuh @greg0ire I rebased this PR. I added some information to changelog and to upgrade file. But I don't find any information how to write upgrade note, so could you please review and give your advice? thank you

greg0ire commented on UPGRADE-3.x.md:
```diff
@@ -1,6 +1,13 @@
+
+## AdminInterface
+
+If you're using a custom implementation of `AdminInterface`, you need to add realisation of `hasAccess` method.
```
This is the UPGRADE note and it is good :)

Soullivaneuh replied:
Except one thing. It should be on `UPGRADE-4.0.md`, the next major.

tim96 replied:
@greg0ire @Soullivaneuh Thank you. Fixed.

greg0ire commented on CHANGELOG.md:
```diff
@@ -62,4 +63,4 @@ to set the CSS class of dropdown item.
 
 ### Removed
-- Remove `btn-outline` from doctrine-orm-admin form actions buttons
+- Remove `btn-outline` from doctrine-orm-admin form actions buttons
```
Please configure your editor to add a newline at end of file  (will make the red pictogram disappear)

tim96 replied:
@greg0ire Thank you. Fixed.

Soullivaneuh commented on UPGRADE-4.0.md:
```diff
@@ -1,6 +1,10 @@
 
+## AdminInterface
+
+If you're using a custom implementation of `AdminInterface`, you need to add realisation of `hasAccess` method.
```
We already have a note about added method on interface here: https://github.com/sonata-project/SonataAdminBundle/pull/3755/files#diff-b2c918cba71bea146d3145ca75754b85R18

Could you complete this part instead? Thanks.

tim96 replied:
@Soullivaneuh Thank you for review. Fixed.

tim96 commented:
In `changelog` format description on their site I saw they using `from` tag. Can we add it too?

Soullivaneuh commented:
Thank you @tim96! :+1:
